src/OAuth2.php

```diff
@@ -1376,23 +1378,60 @@ private function coerceUri($uri)
 
     /**
      * @param string $idToken
-     * @param string|array|null $publicKey
-     * @param array $allowedAlgs
+     * @param Key|Key[]|string|array|null $publicKey
+     * @param string|array $allowedAlg
      * @return object
      */
     private function jwtDecode($idToken, $publicKey, $allowedAlgs)
     {
-        return JWT::decode($idToken, $publicKey, $allowedAlgs);
+        $keys = $this->getFirebaseJwtKeys($publicKey, $allowedAlgs);
+
+        // Default exception if none are caught
+        $e = new \InvalidArgumentException('Key may not be empty');
+        foreach ($keys as $key) {
+            try {
+                return JWT::decode($idToken, $key);
+            } catch (\Exception $e) {
+                // try next alg
+            }
+        }
+        throw $e;
     }
 
-    private function jwtEncode($assertion, $signingKey, $signingAlgorithm, $signingKeyId = null)
+    /**
+     * @return Keys[]
+     */
+    private function getFirebaseJwtKeys($publicKey, $allowedAlgs)
     {
-        return JWT::encode(
-            $assertion,
-            $signingKey,
-            $signingAlgorithm,
-            $signingKeyId
-        );
+        // If $allowedAlgs is empty, assume $publicKey is Key or Key[].
+        if (empty($allowedAlgs)) {
+            return (array) $publicKey;
+        }
+
+        if (is_string($allowedAlgs)) {
```

Comment:
I think `$allowedAlgs` check could instead check against all the supported algos mentioned [here](https://github.com/firebase/php-jwt/blob/main/src/JWT.php#L56-L66).

```
    public static $supported_algs = [
        'ES384' => ['openssl', 'SHA384'],
        'ES256' => ['openssl', 'SHA256'],
        'HS256' => ['hash_hmac', 'SHA256'],
        'HS384' => ['hash_hmac', 'SHA384'],
        'HS512' => ['hash_hmac', 'SHA512'],
        'RS256' => ['openssl', 'SHA256'],
        'RS384' => ['openssl', 'SHA384'],
        'RS512' => ['openssl', 'SHA512'],
        'EdDSA' => ['sodium_crypto', 'EdDSA'],
    ];
```

Reply:
This check is already done in `firebase/php-jwt`, so there is no need to do that check here... we only want to verify type-safety and make sure we handle the types correctly before calling `Firebase\JWT\JWT::decode`.